### PR TITLE
Clean up Visual Studio output and intermediate directories.

### PR DESCRIPTION
**Pull request by haydenroche5**
# Description

Currently, wolfssl.vcxproj and IDE/WIN10/wolfssl-fips.vcxproj do not use the same scheme for their output and intermediate directories. Further, across configuration/platform combinations, wolfssl.vcxproj isn't consistent, either. For example:

```
Release|x64
OutDir: $(SolutionDir)$(Platform)\$(Configuration)\
IntDir: $(Platform)\$(Configuration)\obj\

Release|Win32
OutDir: $(SolutionDir)$(Configuration)\
IntDir: $(Configuration)\obj\
```

This commit makes every configuration/platform combo for all Visual Studio projects follow the same pattern:

```
OutDir: $(SolutionDir)$(Platform)\$(Configuration)\
IntDir: $(Configuration)\$(Platform)\$(ProjectName)_obj\
```

The `$(ProjectName)_obj` piece gets rid of a Visual Studio warning about not mingling the intermediate objects of disparate builds.

# Testing

Tested all config/platform combos on my Windows machine.

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by haydenroche5** [comment]
The "Visual Studio Build Test" failure is inevitable until this is merged and we modify the paths in the batch script driving that PRB.

**Review by kaleb-himes — DISMISSED** [review]
Changes are approved but as you suggested we'll need to update the PRB test before we merge this. That will break all other PR's when we do it so let's wait till after the current release effort (4.5.2?) to merge this. Marking as "Request changes" for now so it doesn't get accidentally merged.

Right now in the test:
```
call :SUB_BUILD_AND_RUN "Debug" "Win32"       "Debug\testsuite.exe"                  "Debug-Win32-output.txt"
call :SUB_BUILD_AND_RUN "Release" "Win32"     "Release\testsuite.exe"                "Release-Win32-output.txt"
call :SUB_BUILD_AND_RUN "DLL Release" "Win32" "DLL Release\testsuite.exe"            "DLL-Release-Win32-output.txt"
call :SUB_BUILD_AND_RUN "DLL Debug" "Win32"   "DLL Debug\testsuite.exe"              "DLL-Debug-Win32-output.txt"
call :SUB_BUILD_AND_RUN "Debug"  "x64"        "x64\Debug\testsuite.exe"        "Debug-x64-output.txt"
call :SUB_BUILD_AND_RUN "Release" "x64"       "x64\Release\testsuite.exe"      "Release-x64-output.txt"
call :SUB_BUILD_AND_RUN "DLL Release" "x64"   "x64\DLL Release\testsuite.exe"  "DLL-Release-x64-output.txt"
call :SUB_BUILD_AND_RUN "DLL Debug" "x64"     "x64\DLL Debug\testsuite.exe"    "DLL-Debug-x64-output.txt"
```

Needs to be updated to:

```
call :SUB_BUILD_AND_RUN "Debug" "Win32"       "Debug\Win32\testsuite.exe"            "Debug-Win32-output.txt"
call :SUB_BUILD_AND_RUN "Release" "Win32"     "Release\Win32\testsuite.exe"          "Release-Win32-output.txt"
call :SUB_BUILD_AND_RUN "DLL Release" "Win32" "DLL Release\Win32\testsuite.exe"      "DLL-Release-Win32-output.txt"
call :SUB_BUILD_AND_RUN "DLL Debug" "Win32"   "DLL Debug\Win32\testsuite.exe"        "DLL-Debug-Win32-output.txt"
call :SUB_BUILD_AND_RUN "Debug" "x64"         "Debug\x64\testsuite.exe"        "Debug-x64-output.txt"
call :SUB_BUILD_AND_RUN "Release" "x64"       "Release\x64\testsuite.exe"      "Release-x64-output.txt"
call :SUB_BUILD_AND_RUN "DLL Release" "x64"   "DLL Release\x64\testsuite.exe"  "DLL-Release-x64-output.txt"
call :SUB_BUILD_AND_RUN "DLL Debug" "x64"     "DLL Debug\x64\testsuite.exe"    "DLL-Debug-x64-output.txt"
```

**Review by ejohnstown — APPROVED** [review]

**Comment by haydenroche5** [comment]
retest this please

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**
